src/java/org/apache/cassandra/service/cleanupstate/CleanupStatePersister.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service.cleanupstate;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.codehaus.jackson.map.ObjectMapper;
+
+public class CleanupStatePersister
+{
+    private static final Logger log = LoggerFactory.getLogger(CleanupStatePersister.class);
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final File persistentFile;
+
+    public CleanupStatePersister(String persistentFileLocation) throws IOException
+    {
+        File cleanupStateFile = new File(persistentFileLocation);
+        try
+        {
+            cleanupStateFile.createNewFile();
+            this.persistentFile = cleanupStateFile;
+        }
+        catch (IOException e)
+        {
+            log.error("Cannot retrieve or create node cleanup state file.", cleanupStateFile.getAbsolutePath(), e);
+            throw e;
+        }
+    }
+
+    public Map<String, Long> readCleanupStateFromFile() throws IOException
```

Comment:
Can we create some typed wrapper class instead of using String? Something like:
```
public class SomeKeyspaceTableWrapper extends Wrapper<String>
{

    private final String keyspace;
    private final String columnFamily;

    public SomeKeyspaceTableWrapper(String keyspace, String columnFamily)
    {
        super(keyspace + ":" + columnFamily);
        this.keyspace = keyspace;
        this.columnFamily = columnFamily;
    }
```
https://github.com/palantir/cassandra/blob/palantir-cassandra-2.2.18/src/java/com/palantir/cassandra/objects/Wrapper.java